constrained-voucher.mkd

```diff
@@ -646,36 +646,52 @@ INSERT_FIG_FROM_FILE examples/voucher-example1.txt END
+    h'1234....5678', #voucher-request binary content (CBOR)
+    h'4567....1234'  #voucher-request binary public signature
   ]
 )
 ~~~~
-{: #fig-cose title='cose-sign1 example in CBOR diagnostic notation' align="left"}
-
-The {{COSE-registry}} specifies the integers that replace the strings and the mnemonics in {{fig-cose}}.
-The value of the key identifier "kid" parameter is an example value.
-Usually a hash of the public key is used to identify the public key. The choice of key identifier method is
-vendor-specific.
-The public key and its hash are derived from the relevant certificate (Pledge, Registrar or MASA certificate).
-[TBD EDNOTE: how can MASA know which kid method the Registrar has used/supports? Does it matter?]
-
-In {{cosesign}} a binary cose-sign1 object is shown based on the voucher-request example of {{example2}}.
+{: #fig-cose title='COSE_Sign1 example in CBOR diagnostic notation' align="left"}
+
+The {{COSE-registry}} specifies the integers/encoding for the "alg" and "kid" fields in {{fig-cose}}. The "alg" 
+field restricts the key usage for verification of this COSE object to a particular cryptographic algorithm.
+
+The "kid" field is optionally present: it is an unprotected field that identifies the public key of the key pair that was used to sign this 
+message. The value of the key identifier "kid" parameter is an example value.
+Usually a hash of the public key is used to identify the public key, but a device serial number may also be used. The choice of key identifier method is
+vendor-specific. If "kid" is not present, then a verifying party needs to use other context information to 
+retrieve the right public key to verify the COSE_Sign1 object against. For example, this context information 
+may be a unique serial number encoded in the binary content (CBOR) field.
+
+A Registrar MAY use a "kid" parameter in its RVR to identify its signing key as used to sign the RVR. 
+The method of generating this "kid" is vendor-specific and SHOULD be configurable in the Registrar to 
+support commonly used methods. A Registrar SHOULD have a configuration option to set the "kid" to a particular 
+binary value. By default, a Registrar SHOULD NOT include a "kid" parameter in its RVR since the signing key 
+is already identified by the included signing certificates in the COSE "x5bag" structure.
+
+A Pledge normally SHOULD NOT use a "kid" parameter in its PVR, because its signing key is already identified 
+by the Pledge's unique serial number that is included in the PVR. Still, where needed the Pledge MAY use 
+a "kid" parameter in its PVR to help the MASA identify the right public key to verify against. This can occur 
+for example if a Pledge has multiple IDevID identities.
+A Registrar normally SHOULD ignore a "kid" parameter used in a received PVR, as this information is intended for the MASA. 
+In other words, there is no need for the Registrar to verify the contents of this field. 
```

Comment:
```suggestion
A Registrar normally SHOULD ignore a "kid" parameter used in a received PVR, as this information is intended for the MASA. 
In other words, there is no need for the Registrar to verify the contents of this field, but it may include it in an audit log.
```